src/torrent.cpp

```diff
@@ -12310,7 +12312,10 @@ namespace {
 		{
 			announce_with_tracker(r.event);
```

Comment:
if this call fails, `update_tracker_timer()` won't be called, with your patch. I'm not sure exactly what the consequence of that is, but it's possible the tracker will be tried over and over again in quick succession.
I would feel better about this change if you would check the return value here and call `update_tracker_timer()` if it fails. Or perhaps even better, make `announce_with_tracker()` update the timer unconditionally.

Reply:
`torrent::announce_with_tracker` returns `void`, so there's no way to know if it fails.

In all cases where `torrent::announce_with_tracker` returns early (fails), there are either no trackers, or libtorrent isn't announcing. In both cases, `torrent::update_tracker_timer` would do nothing anyway, so I don't think it matters.